cloudinit/net/__init__.py

```diff
@@ -830,6 +842,21 @@ def find_mac(flat_list):
     return results
 
 
+def get_interfaces_by_mac_on_netbsd():
+    ret = {}
+    re_field_match = (
+            r"(?P<ifname>\w+).*address:\s"
+            r"(?P<mac>([\da-f]{2}[:-]){5}([\da-f]{2})).*")
+    (out, _) = util.subp(['ifconfig', '-a'])
+    if_lines = re.sub(r'\n\s+', ' ', out).splitlines()
```

Comment:
and this type of flatten cannot be applied in `get_interfaces_by_mac_on_freebsd()` above?

Reply:
The ifconfig outputs are different enough to justify two different functions. OpenBSD is slightly different too.